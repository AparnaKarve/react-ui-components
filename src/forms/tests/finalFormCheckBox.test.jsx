import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import { FinalFormCheckBox } from '../';
import { fieldInputProp, fieldMetaProps } from './mocks';

describe('Final form checkbox component', () => {
  const initialProps = {};
  beforeEach(() => {
    initialProps.meta = { ...fieldMetaProps };
    initialProps.input = { ...fieldInputProp };
    initialProps.label = 'foo';
    initialProps.validateOnMount = true;
  });

  it('Should render correctly', () => {
    const tree = renderer.create(<FinalFormCheckBox {...initialProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should render with error message', () => {
    const props = { ...initialProps, meta: { ...initialProps.meta, error: 'Error message' } };
    const tree = renderer.create(<FinalFormCheckBox {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should call onChange event', () => {
    const onChange = jest.fn();
    const props = { ...initialProps, input: { ...initialProps.input, onChange } };
    const wrapper = mount(<FinalFormCheckBox {...props} />);
    wrapper.find('input[type="checkbox"]').simulate('change');
    expect(onChange).toHaveBeenCalled();
  });
});
