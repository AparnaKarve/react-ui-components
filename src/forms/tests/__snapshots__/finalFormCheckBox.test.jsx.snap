// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Final form checkbox component Should render correctly 1`] = `
<div
  className="form-group"
>
  <label
    className="control-label col-xs-2"
  >
    foo
  </label>
  <div
    className="col-xs-8"
  >
    <div
      className="checkbox"
      style={undefined}
    >
      <label
        title=""
      >
        <input
          disabled={false}
          name="name"
          onBlur={[MockFunction]}
          onChange={[MockFunction]}
          onFocus={[MockFunction]}
          type="checkbox"
          value=""
        />
      </label>
    </div>
  </div>
</div>
`;

exports[`Final form checkbox component Should render with error message 1`] = `
<div
  className="form-group has-error"
>
  <label
    className="control-label col-xs-2"
  >
    foo
  </label>
  <div
    className="col-xs-8"
  >
    <div
      className="checkbox"
      style={undefined}
    >
      <label
        title=""
      >
        <input
          disabled={false}
          name="name"
          onBlur={[MockFunction]}
          onChange={[MockFunction]}
          onFocus={[MockFunction]}
          type="checkbox"
          value=""
        />
      </label>
    </div>
    <span
      className="help-block"
    >
      Error message
    </span>
  </div>
</div>
`;
